### DISCUSSION TITLE
pyvista save slice while preserving dimensions

### DISCUSSION THREAD

**Discussion by vmeline** [pyvista/pyvista#1880] [Q&A] [answered]
Hello, 
I am trying to save the slices of a mesh as 2D images using pyvista mesh.slice_along_axis()
However, it seems that the saved slices do not conserve the initial dimensions 
Should I add a uniform.grid ? If so can you help me with this ?

An example of the code that I am using, ,thank you for your help:

```py
#read and display mesh
mesh = pv.read('....stl')
cpos = mesh.plot(cpos="xy", show_edges=False)

# get slices of the mesh
zslices = mesh.slice_along_axis(n=250, axis='y')
zslices.plot(color='white', opacity=1, show_grid=True)

#save the slices
for i, slc in enumerate(zslices):
    p = pv.Plotter(off_screen=True, notebook=False)
    p.add_mesh(zslices[0], color='white', opacity=1)
    p.view_zx(negative=False)
    p.show(screenshot='C:/Users/vmeline/Desktop/New folder/slice_%d.png' % i)
    p.store_image = True
    p.close()

```

**Comment by banesullivan** [comment, marked as answer]
Your best bet, if you want to preserve the dimensions of the mesh is to use a UniformGrid and the special methods around it for subsampling to get your slices.

With that said, these may also be helpful for you: 
- https://github.com/pyvista/pyvista-support/issues/70
- https://github.com/pyvista/pyvista-support/issues/89

**Comment by RichardScottOZ** [comment]
and something like this :
```python
#varying for cells/points  -1 or not at the end of the reshape
reshaped_arr = slices['scalar_data'].reshape(mesh.dimensions[0]-1, mesh.dimensions[1]-1) 
```
